src/OpenLoco/Vehicles/VehicleHead.cpp

```diff
@@ -1752,10 +1752,17 @@ namespace OpenLoco::Vehicles
     // 0x0042750E
     void VehicleHead::produceLeavingAirportSound()
     {
-        // Creates a random sound
-        registers regs;
-        regs.esi = reinterpret_cast<int32_t>(this);
-        call(0x0042750E, regs);
+        Vehicle train(this);
+        auto* vehObj = train.cars.firstCar.body->object();
+        if (vehObj != nullptr && vehObj->numStartSounds != 0)
+        {
+
+            auto randSoundIndex = gPrng().randNext(vehObj->numStartSounds & NumStartSounds::mask);
```

Comment:
How does this work? What do `vehObj->numStartSounds` and `NumStartSounds::mask` contain?

Reply:
The mask is because the highest order bit is used to indicate if there is an approaching whistle at the end of the start sounds.